R/gg-plots.R

```diff
@@ -242,74 +256,128 @@ ggally_density <- function(data, mapping, ...){
 ggally_cor <- function(
   data,
   mapping,
-  alignPercent = 0.6,
-  method = "pearson", use = "complete.obs",
-  corAlignPercent = NULL, corMethod = NULL, corUse = NULL,
-  displayGrid = TRUE,
-  ...
-){
+  ...,
+  stars = TRUE,
+  method = "pearson",
+  use = "complete.obs",
+  displayGrid = FALSE,
+  digits = 3,
+  title_args = list(...),
+  group_args = list(...),
+  justify_labels = "right",
+  alignPercent = 0.5,
+  title = "Corr"
+) {
+  ggally_exec_text(
+    data = data,
+    mapping = mapping,
+    alignPercent = alignPercent,
+    use = use,
+    displayGrid = displayGrid,
+    title_args = title_args,
+    group_args = group_args,
+    justify_labels = justify_labels,
+    justify_text = "left",
+    sep = if ("colour" %in% names(mapping)) ": " else ":\n",
+    title = title,
+    text_fn = function(x, y) {
+      if (is_date(x)) {
+        x <- as.numeric(x)
+      }
+      if (is_date(y)) {
+        y <- as.numeric(y)
+      }
 
-  if (! is.null(corAlignPercent)) {
-    stop("'corAlignPercent' is deprecated.  Please use argument 'alignPercent'")
-  }
-  if (! is.null(corMethod)) {
-    stop("'corMethod' is deprecated.  Please use argument 'method'")
-  }
-  if (! is.null(corUse)) {
-    stop("'corUse' is deprecated.  Please use argument 'use'")
-  }
+      corObj <- stats::cor.test(x, y, method = method, use = use)
+
+      # make sure all values have X-many decimal places
+      cor_est <- as.numeric(corObj$estimate)
+      cor_txt <- formatC(cor_est, digits = digits, format = "f")
+
+      # if stars should be added
+      if (isTRUE(stars)) {
+        cor_stars <- function(pValue) {
+          if (pValue < 0.001) {
+            " ***"
+          } else if (pValue < 0.01) {
+            " **"
+          } else if (pValue < 0.05) {
+            " *"
+          } else if (pValue < 0.10) {
+            " ."
+          } else {
+            ""
+          }
+        }
+        cor_txt <- str_c(
+          cor_txt,
+          cor_stars(corObj$p.value)
+        )
+      }
 
-  useOptions <- c(
-    "all.obs",
-    "complete.obs",
-    "pairwise.complete.obs",
-    "everything",
-    "na.or.complete"
+      cor_txt
+    }
   )
-  use <- pmatch(use, useOptions)
-  if (is.na(use)) {
-    warning("correlation 'use' not found.  Using default value of 'all.obs'")
-    use <- useOptions[1]
-  } else {
-    use <- useOptions[use]
-  }
+}
 
-  cor_fn <- function(x, y) {
-    # also do ddply below if fn is altered
-    cor(x, y, method = method, use = use)
+
+#' Generalized Text Display
+#'
+#' @param data data set using
+#' @param mapping aesthetics being used
+#' @param title title text to be displayed
+#' @param text_fn function that takes in \code{x} and \code{y} and returns a text string
+#' @param use \code{use} supplied to \code{\link[stats]{cor}} function
```

Comment:
This is a specfic parameter for cor only. Could be remplaced by a `complete.cases` TRUE or FALSE to be more generic